### PR TITLE
Migrate from Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by jlundqvist-criteo**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 8
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'adopt'
```
Comment (gray-matter) [inline .github/workflows/deploy.yml:19]: The other file says "cache: maven", any reason for doing that there and not here?

File: .travis.yml [hunk 2]
```diff
@@ -1,19 +0,0 @@
...
-env:
-  global:
-    secure: o2HyqCaCFHRTkncncfKJPxgkdTm2qG/+qjs0cu55TobbgLaTW8kNJYEAVaHTpKFylWaSLplPmD+FbP788AFtmVMGRK5Z+b1+zeZsDTLUeli+9DxpOwdpveM97E9zFwyioT+glaAsVvxsWgX7i+yw9Z23BDVBcC50x5E1huOjBl0U4zyrrPNL/sjv5EjQ5IYrq3HNcUQFdUgyE6pWtN2ibBW5jwycRKYyjsGXnF6htWb4yBdf78rmaWblV4fRvVtlJJj+e+WFnYhr+s3Fb3roG1BkujWxQ3AtlEXDoaLrvKqGzClsDQlcbF0Dffl1xLAK3fIFuX3GOwBPg4rAEttM6zGcNOZXIh/+uGyN238bOs1q5h+S80SsQsqO32Ox34EhIcXDWIxg4C9ZE0NMtPts/kd8vGfld6p7yImHR+tcOe+HEKpF0lWSPirVtZNRkjZRQ9ZnGI02p9Pug9OX42Ux3hukS3JzYXYujLaNS8tb29Q9pg0IISUJkRNlDEeRK+H4y0D4eGiuxGPDFS09a8NhCBcI8OwFLAewckuoTD46RaoqazlFnsJc1mUumejbZhHFx8ZTdG2k8QZq51oiN3kvZjZSRgxTf19zNEyjg7uko5lnu/P2o6BcX/yrdVPTvUztl1du8lz/F6UNxp4SUadnwUJWI/Xow+izoFuOM2ROfy8=
-before_deploy:
-- openssl aes-256-cbc -K $encrypted_91ee6a0187b8_key -iv $encrypted_91ee6a0187b8_iv
-  -in .travis/keys.tar.enc -out .travis/keys.tar -d
-- tar xvf .travis/keys.tar
-deploy:
-  provider: script
-  script: bash scripts/deploy.sh
```
Comment (gray-matter) [inline .travis.yml:15]: Expecting to see a removal of this file

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          MAVEN_PUBRING_GPG_BASE64: ${{ secrets.MAVEN_PUBRING_GPG_BASE64 }}
+          MAVEN_SECRING_GPG_BASE64: ${{ secrets.MAVEN_SECRING_GPG_BASE64 }}
+        run: echo $MAVEN_PUBRING_GPG_BASE64 > pubring.gpg && echo $MAVEN_SECRING_GPG_BASE64 > secring.gpg
+      - name: Deploy on Sonatype
+        env:
+          SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
+          MAVEN_SECRING_PASSWORD: ${{ secrets.MAVEN_SECRING_PASSWORD }}
+        run: mvn deploy -P release --settings .github/mvn_settings_deploy.xml -DskipTests
+      - name: Clean credentials
+        if: always()
```
Comment (gray-matter) [inline .github/workflows/deploy.yml:31]: Is this line necessary?

**Review by gray-matter — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (jlundqvist-criteo) [reply .github/workflows/deploy.yml:31]: Could you specify which line you mean?

File: .travis.yml (hunk 2 above)
Comment (jlundqvist-criteo) [reply .travis.yml:15]: Good catch, will fix

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (jlundqvist-criteo) [reply .github/workflows/deploy.yml:19]: No, not really. I'll add it.

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (jlundqvist-criteo) [reply .github/workflows/deploy.yml:31]: if: always() Is needed to remove the secrets files even if a previous step fails

Comment (gray-matter) [reply .github/workflows/deploy.yml:31]: The comment always refers to the line right above. Thanks for the piece of info

**Comment by gray-matter** [comment]
Also, the output says "continuous-integration/travis-ci Expected — Waiting for status to be reported" which, of course, won't happen. How does one migrate this part to GH actions?

**Comment by jlundqvist-criteo** [comment]
> 

I am not sure, I will investigate. When trying out the PR against the master branch of my forked copy, it ran the new github actions build step and not the old travis one.

**Comment by jlundqvist-criteo** [comment]
> Also, the output says "continuous-integration/travis-ci Expected — Waiting for status to be reported" which, of course, won't happen. How does one migrate this part to GH actions?

This should happen automatically (it did for the PR to my own garmadon fork's master branch), I could not really see anthing in the project settings that suggests otherwise but I will investigate

**Comment by jlundqvist-criteo** [comment]
> Also, the output says "continuous-integration/travis-ci Expected — Waiting for status to be reported" which, of course, won't happen. How does one migrate this part to GH actions?

After investigating and comparing to other repos that have migrated (i.e. cuttle) still have not found why the new CI does not happen. My best guess is that I first need an approval from someone else in the team. I'd like to go ahead even without the CI running to see what happens, and then create a test PR to verify it does run as expected.

**Review by gray-matter — APPROVED** [review]

**Pull request merged**